src/components/Drawer/Drawer.js

```diff
@@ -1,55 +1,112 @@
+
+    useEffect(() => {
+        if (isExpandProps !== null) {
+            setIsExpand(isExpandProps);
+        }
+    }, [isExpandProps]);
+
+    function getStatus() {
+        const STATUS_LIST = {
+            IS_SHOWN_AND_EXPENDED: 'isShownAndExpanded',
+            IS_SHOWN_AND_CLOSED: 'isShownAndClosed',
+            EMPTY: '',
+        };
+
+        if (isShow) {
+            return isExpand ? STATUS_LIST.IS_SHOWN_AND_EXPENDED : STATUS_LIST.IS_SHOWN_AND_CLOSED;
+        }
+
+        return STATUS_LIST.EMPTY;
+    }
 
     function handleClick() {
-        setIsOpen(!isOpen);
+        if (onClick) {
+            onClick();
+        } else {
+            setIsExpand(!isExpand);
+        }
     }
 
-    return (
-        <div {...block({ isOpen })} {...rest}>
-            <header {...elem('head', props)}>
-                <h4 {...elem('headTitle', props)}>{title}</h4>
-                <div {...elem('headButtons', props)}>
-                    <button
-                        {...elem('extendButton', { isOpen })}
-                        type="button"
-                        onClick={handleClick}
-                    >
-                        <IoIosArrowUp {...elem('extendIcon', props)} />
-                    </button>
+    return ReactDOM.createPortal(
+        <Transition appear unmountOnExit in={isShow} timeout={TRANSITION_DURATION}>
+            {state => (
+                <div
+                    {...block({ [getStatus()]: state !== 'entering' })}
+                    aria-hidden={!isShow}
+                    {...rest}
+                >
+                    <header {...elem('head', props)}>
+                        <h4 {...elem('headTitle', props)}>{title}</h4>
+                        <div {...elem('headButtons', props)}>
+                            <button
+                                {...elem('expandButton', { isExpanded: isExpand })}
+                                type="button"
+                                onClick={handleClick}
+                            >
+                                <IoIosArrowUp {...elem('expandIcon', props)} />
+                            </button>
+                        </div>
+                    </header>
+                    <section {...elem('body', props)} aria-hidden={!(isShow && isExpand)}>
+                        {children}
+                    </section>
                 </div>
-            </header>
-            <section {...elem('body', props)} aria-hidden={!isOpen}>
-                {children}
-            </section>
-        </div>
+            )}
+        </Transition>,
+        document.body
     );
 };
 
 Drawer.propTypes = {
     /** Drawer title */
     title: PropTypes.string.isRequired,
     /** Set open/close status for initial state */
-    initialIsOpen: PropTypes.bool,
+    initialIsExpand: PropTypes.bool,
+    /** Control visibility Drawer component */
+    isShow: PropTypes.bool,
```

Comment:
`isShow` and `isExpand` is bad English, and hence hard to read. It should be `doShow` and `doExpand` or `isShown` and is `Expanded` instead. That applies to several places, not just here (e.g. intialIsExpand should be initiallyDoExpand, etc).